contributors/devel/sig-architecture/api_changes.md

```diff
@@ -744,6 +744,16 @@ doing!
 Check out the [E2E docs](../sig-testing/e2e-tests.md) for detailed information about how to
 write end-to-end tests for your feature.
 
+When introducing a new API or field as an alpha feature, you have to add a
+[Feature label](../sig-testing/e2e-tests.md#kinds-of-tests). This prevents the
+test from running in the main suites, which have the feature disabled by default.
+You can choose to opt-in the test for the alpha suites, like in this
+[example](https://github.com/kubernetes/test-infra/pull/20833/files).
+
+When graduating to beta and making the feature enabled by default, you have
+to remove the label from the test so that it runs in the main suite. Don't
+forget to remove the label from the alpha suites.
```

Comment:
Linking to the kinds-of-tests bit and mentioning the consideration about feature-gated non-default features is good, but I would limit the content here to saying "make sure the tests for features which are enabled by default are running in default e2e suites".

Separately, we should work with sig-testing to see if we can improve the current approach to eliminate some of the current problems:
* test/feature authors have to remember to remove tags from test names when defaulting the feature on. Ideally, tests would be decorated with something that indicates both the feature name AND whether it is on by default, and we could run a programmatic check that would flag tests whose tag does not match the feature default state (e.g. `[Feature:MyFeature][Default:Enabled]`)
* features that are enabled by default but are not yet GA get feature tags removed from their test name in order to be included in default suites. On clusters where a beta feature was disabled because it was problematic, this means the e2e test starts failing and there isn't a crisp way to skip the test. Ideally, we'd keep the `Feature:$name` tag in the test name until the feature gate is GA and locked on, so a cluster that intentionally disables beta feature `MyFeature` could also skip `[Feature:MyFeature]` tests

Reply:
I suspect those conversation will take a while.
In the meantime, don't you think it's worth being more explicit about what should be done? Otherwise it might take contributors a lot of digging to understand it.

Reply:
I'd like to avoid denormalizing guidance I expect to change in the future... if we link to the testing page and describe the desired *outcome* here ("make sure the e2e tests are running in default presubmits for a feature that is getting promoted to beta and enabled by default") that seems ok

Reply:
Updated